node/core/dispute-coordinator/src/real/ordering/mod.rs

```diff
@@ -295,12 +295,13 @@ impl OrderingProvider {
 
 				ancestors.push(*hash);
 			}
+
 			match hashes.last() {
-				Some(last_hash) => {
+				Some(last_hash) if !block_numbers.is_empty() => {
```

Comment:
I would find hashes.len() more intuitive here and would also help avoid the clone above.

Reply:
Cloning is the same as copying two integers.
The idea of the check is that while `hashes` is not empty, saturating sub could produce empty block numbers range and we don't want to go further (since, well, we don't have block numbers).

Anyway, needs to be reworked together with chain api

Reply:
fair point!